### PR TITLE
Accelerate Answer Finding with LLM-Powered Website Analysis

### PR DESCRIPTION
**Pull request by bipinbids**

### CODE REVIEW

File: chat_with_website/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Website Guru
+
+Website Guru is a Streamlit application that allows you to interact with and ask questions about a given website using advanced language models. The app uses a combination of document loading, text splitting, vector storage, and retrieval chains to generate accurate responses based on the content of the specified website.
```
Comment (hozen-groq) [inline chat_with_website/README.md:3]: Suggest replacing "advanced language models" with Large Language Models (LLMs).

Comment (hozen-groq) [inline chat_with_website/README.md:3]:
```suggestion
Website Guru is a Streamlit application that allows you to interact with and ask questions about a given website using Large Language Models (LLMs). The app uses a combination of document loading, text splitting, vector storage, and retrieval chains to generate accurate responses based on the content of the specified website.

```

File: chat_with_website/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+## Usage
+
+1. Run the Streamlit app:
+
+	streamlit run app.py
+
+2. Open your web browser and go to http://localhost:8501.
+
+3. Enter the website address you want to analyze and choose a language model.
```
Comment (hozen-groq) [inline chat_with_website/README.md:40]:
```suggestion
3. Enter the website address you want to analyze and select LLM of your choice.

```

**Comment by hozen-groq** [comment]
Hi, Bipin! Thank you so much for sharing this cool project and for wanting to contribute to our Groq API Cookbook! 🤩 I've left a couple suggestions for edits in the code and also have a question before we can merge this PR. I tried the Streamlit app (very cool), but I was wondering what the delay in response is? Is it possible that the chunking method also takes a second or two in the backend? I think it would be great if we could add a timer and something to the UI that indicates how much time the inference portion takes with Groq!

**Comment by bipinbids** [comment]
Hello Hatice, yes the chucking is taking some time, let me display a timer in the UI for the chunking and the inference, thanks for your valuable feedback, this has been an exciting first open-source contribution.

**Comment by hozen-groq** [comment]
Hi, Bipin! I just ran the app again to see the changes you made - they look great! Upon further inspection, I think that the right home for your app would be at https://console.groq.com/docs/showcase-applications instead of the cookbook since the cookbook is more for step-by-step tutorials that explain the code and what's being done. If you could add a ["powered by Groq" badge to your app](https://photos.google.com/share/AF1QipNbkJWYjEBMb5cwugnE9oYXUXlfTVsqFUxypbpd0S8XoX6ldnpY6mkddmo6idZ07g?key=Vkx3TzVQLVZxX1NSTDl4ck14WFlGOVdaV1lWdkt3), I could go in and edit out docs to include this app!

**Comment by bipinbids** [comment]
That sounds like a plan @hozen-groq, let's do that, I have added the badge, let me know if you want it to be an image.

File: chat_with_website/app.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+# First column: Input for website address
+with c1:
+    website_add = st.text_input("Enter Website:", value="https://en.wikipedia.org/wiki/Bigfoot")
+
+# Second column: Selectbox for choosing one of the available models
+with c2:
+    llm_model_name = st.selectbox(
+        "What model would you like to use?",
+        ("mixtral-8x7b-32768", "llama3-8b-8192", "llama3-70b-8192", "gemma-7b-it", "gemma2-9b-it")
```
Comment (ozenhati) [inline chat_with_website/app.py:42]:
```suggestion
        ("meta-llama/llama-4-scout-17b-16e-instruct", "llama-3.3-70b-versatile", "gemma2-9b-it")

```

File: chat_with_website/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+streamlit
+langchain
+langchain_groq
+langchain_community
+sentence-transformers
+faiss-cpu
+python-dotenv
```
Comment (ozenhati) [inline chat_with_website/requirements.txt:7]:
```suggestion
python-dotenv
bs4
```

**Comment by ozenhati** [comment]
@bipinbids Hi, Bipin! Apologies for such a delay in response - upon further inspection, I've decided to add this to our cookbook as well as we're going to revamp the apps showcase and I think that this is a great addition for those who want to create apps with vector stores. I appreciate your contribution and hope that I'll see more of your work here! :)

**Pull request closed without merging**